examples/matmul_heuristic_plugin/CMakeLists.txt

```diff
@@ -0,0 +1,30 @@
+# SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+# All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause
+cmake_minimum_required(VERSION 3.10 FATAL_ERROR)
+project(nvfuser_matmul_heuristic_plugin LANGUAGES CXX)
+
+set(CMAKE_CXX_STANDARD 17)
+
+set(nvfuser_include_dir /opt/pytorch/nvfuser/csrc CACHE FILEPATH REQUIRED)
+set(nvfuser_lib_dir /opt/pytorch/nvfuser/build CACHE FILEPATH REQUIRED)
```

Comment:
Building the plugin does not require nvFuser, correct? I assume people can just download the `csrc/scheduler/matmul_heuristic_plugin_api.h` file and use it. If this is the recommended way for people to write plugin, should we not hardcode anything about nvFuser here, and just create a symlink here pointing to `csrc/scheduler/matmul_heuristic_plugin_api.h`?

Reply:
Actually, you're right that we can compile by just symlinking the header here but we also need to link the library. If we don't link against `nvfuser_codegen` then we cannot `dlsym` the function as there will be missing symbols. I had thought this would work by finding the symbols in the executable, but it does not work that way apparently. I think ideally we'd install a `FindnvFuser.cmake` somewhere so that we could just `find_package(Nvfuser REQUIRED);` then use `${NVFUSER_INCLUDE_DIR}` and `${NVFUSER_LIBS}` etc. Currently `find_package(Nvfuser)` works but it does not export the installation include or lib dirs or the `nvfuser_codegen` lib as variables (you have to hard-code the lib name). That is probably a topic for another issue.

Reply:
Ah, I figured this out. The `sinh_libtorch` example has a proper cmake setup that should work for this example as well. I was hitting problems with missing headers but that is because I had not re-run `python setup.py install`, which populates the include dir `/opt/pytorch/nvfuser/nvfuser/include/nvfuser/`. I think once we install the package it works as expected.

Reply:
Why does the plugin has to link nvFuser? I am doing experiment here: https://github.com/zasdfgbnm/dlopen-polymorphic-class
seems that it is not required.

Plugins linking with nvFuser could have versioning issues? For example, if I created my `libplugin.so` and link with nvFuser 1.1.1, will nvFuser 1.1.2 be able to use that plugin?